framework/src/vectorpostprocessors/PointSamplerBase.C

```diff
@@ -159,10 +170,44 @@ PointSamplerBase::transferPointsVector(std::vector<Point> && points)
 const Elem *
 PointSamplerBase::getLocalElemContainingPoint(const Point & p)
 {
-  const Elem * elem = (*_pl)(p);
+  const Elem * elem;
+  if (_discontinuous_variables)
+  {
+    // Get all possible elements the point may be in
+    std::set<const Elem *> candidate_elements;
+    (*_pl)(p, candidate_elements);
+
+    if (candidate_elements.size())
+    {
+      // Look at all the element IDs
+      std::set<dof_id_type> candidate_ids;
+      std::transform(candidate_elements.begin(),
+                     candidate_elements.end(),
+                     std::inserter(candidate_ids, candidate_ids.end()),
+                     [](const Elem * elem) { return elem->id(); });
+
+      // Process owning the global minimum owns the point
+      comm().set_union(candidate_ids);
+      
+      // If we know of the minimum, this'll be valid. Otherwise, it'll be
+      // nullptr which is fine
+      elem = _mesh.queryElemPtr(*(candidate_ids.begin()));
+ 
+      // Print a warning if it's on a face and a variable is discontinuous
+      if (_warn_discontinuous_face_values && candidate_ids.size() > 1)
+        mooseWarning("A discontinuous variable is sampled on a face, at ", p);
+
+      }
+    }
+  }
+  else // continuous variables
+    elem = (*_pl)(p);
 
   if (elem && elem->processor_id() == processor_id())
     return elem;
+    
+  return nullptr;
+}
```

Comment:
I think I messed this up a little :D

Reply:
hitting this bad boy now too:

/home/giudguil/projects/moose/framework/src/vectorpostprocessors/PointSamplerBase.C:202:33: warning: 'elem' may be used uninitialized in this function [-Wmaybe-uninitialized]
  202 |   if (elem && elem->processor_id() == processor_id())